core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -122,11 +116,14 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def fromConfigObject(co: ConfigObject, default: Option[V] :: U): Try[FieldType[K, V] :: T] = {
       val keyStr = mapping(key.value.toString().tail)
       for {
-        v <- improveFailure(
-          vFieldConvert.value.from(co.get(keyStr)) match {
-            case Failure(CannotConvertNullException) =>
+        v <- improveFailure[V](
+          (co.get(keyStr), vFieldConvert.value) match {
+            case (null, converter: OptionConfigConvert[_]) =>
```

Comment:
I like this idea, what do you think @jcazevedo ?
FYI an alternative could be to break the API again and pass an `Option[ConfigValue]` to `.from` to help the user understand that a key could be not found. I'm not sure about this approach because in most of the cases I expect a missing key to be an error, but it's a possibility.